packages/core/src/data_sources/model/conditional_variables/ComponentDataCondition.ts

```diff
@@ -1,61 +1,130 @@
-        droppable: false,
+  get defaults(): ComponentDefinitionDefined {
+    return {
+      // @ts-ignore
+      ...super.defaults,
+      droppable: false,
+      type: DataConditionType,
+      dataResolver: {
+        condition: {
+          left: '',
+          operator: StringOperation.equalsIgnoreCase,
+          right: '',
+        },
       },
-      opt,
-    );
-    this.dataResolver = dataConditionInstance;
-    this.dataResolver.onValueChange = this.handleConditionChange.bind(this);
+      components: [
+        {
+          type: DataConditionIfTrueType,
+        },
+        {
+          type: DataConditionIfFalseType,
+        },
+      ],
+    };
   }
 
-  getCondition() {
-    return this.dataResolver.getCondition();
+  constructor(props: ComponentDataConditionProps, opt: ComponentOptions) {
+    // @ts-ignore
+    super(props, opt);
+
+    const { condition } = props.dataResolver;
+    this.dataResolver = new DataCondition({ condition }, { em: opt.em });
+
+    this.listenToPropsChange();
   }
 
-  getIfTrue() {
-    return this.dataResolver.getIfTrue();
+  isTrue() {
+    return this.dataResolver.isTrue();
   }
 
-  getIfFalse() {
-    return this.dataResolver.getIfFalse();
+  getCondition() {
+    return this.dataResolver.getCondition();
   }
 
-  private handleConditionChange() {
-    this.components(this.dataResolver.getDataValue());
+  getIfTrueContent(): Component | undefined {
+    return this.components().at(0);
   }
 
-  static isComponent(el: HTMLElement) {
-    return toLowerCase(el.tagName) === DataConditionType;
+  getIfFalseContent(): Component | undefined {
+    return this.components().at(1);
+  }
+
+  getOutputContent(): Component | undefined {
+    return this.isTrue() ? this.getIfTrueContent() : this.getIfFalseContent();
   }
 
   setCondition(newCondition: ConditionProps) {
     this.dataResolver.setCondition(newCondition);
   }
 
-  setIfTrue(newIfTrue: any) {
-    this.dataResolver.setIfTrue(newIfTrue);
+  setIfTrueContent(content: ComponentAddType): Component | undefined {
+    return this.replaceChildAtIndex(0, content);
+  }
+
+  setIfFalseContent(content: ComponentAddType): Component | undefined {
+    return this.replaceChildAtIndex(1, content);
   }
 
-  setIfFalse(newIfFalse: any) {
-    this.dataResolver.setIfFalse(newIfFalse);
+  getInnerHTML(opts?: ToHTMLOptions): string {
+    return this.getOutputContent()?.getInnerHTML(opts) ?? '';
   }
 
-  toJSON(): ComponentDefinition {
-    return this.dataResolver.toJSON();
+  private replaceChildAtIndex(index: number, newContent: ComponentAddType): Component | undefined {
+    const components = this.components();
+    const existingComponent = components.at(index);
+
+    if (existingComponent) {
+      this.components().remove(existingComponent);
+      const newComponents = this.append(newContent, { at: index })[0];
+      this.trigger(DataConditionOutputChangedEvent);
+      return newComponents;
+    }
+  }
```

Comment:
Not sure about this. I think we should not allow removing the already defined true/false.
I think you just need to update those components.
```js
existingComponent?.components(newContent);
```